src/diff.c

```diff
@@ -654,10 +654,13 @@ diff_trace_origin(struct view *view, struct line *line)
 
 	for (; diff < line && !file; diff++) {
 		const char *data = box_text(diff);
-
-		if (!prefixcmp(data, "--- a/")) {
-			file = data + STRING_SIZE("--- a/");
-			break;
+		if (!prefixcmp(data, "--- ")) {
+			const char *data_n1 = box_text(diff+1);
+			const char *data_n2 = box_text(diff+2);
+			if(!prefixcmp(data_n1, "+++ ") && !prefixcmp(data_n2, "@@ ")){
+				file = data + STRING_SIZE("--- ");
```

Comment:
all these hardcoded strings can probably be from https://github.com/jonas/tig/blob/master/include/tig/line.h#L29